curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentNode.java

```diff
@@ -462,7 +484,12 @@ else if ( existingPath != null && mode.isSequential() )
             if ( localCreateMethod == null )
             {
                 CreateBuilderMain createBuilder = mode.isTTL() ? client.create().withTtl(ttl) : client.create();
-                CreateModable<ACLBackgroundPathAndBytesable<String>> tempCreateMethod = useProtection ? createBuilder.creatingParentContainersIfNeeded().withProtection() : createBuilder.creatingParentContainersIfNeeded();
+                CreateModable<ACLBackgroundPathAndBytesable<String>> tempCreateMethod;
+                if (useParentCreation) {
+                    tempCreateMethod = useProtection ? createBuilder.creatingParentContainersIfNeeded().withProtection() : createBuilder.creatingParentContainersIfNeeded();
+                } else {
+                    tempCreateMethod = useProtection ? createBuilder.withProtection() : createBuilder;
+                }
                 createMethod.compareAndSet(null, tempCreateMethod);
```

Comment:
`createMethod` as `AtomicReference` or even `volatile` is overkill...we might review this part in another pass.

Reply:
@TisonKun, you're suggesting just moving the createMethod to be a member variable that is initialised in the constructor? I can't see why it's lazily instantiated currently, possibly for historical reasons.

It can probably be addressed as a different PR?

Reply:
Yes it could be addressed as a different PR.